modules/color-picker/react/lib/ColorInput.tsx

```diff
@@ -44,23 +44,44 @@ const CustomHexInput = styled(TextInput)<Pick<ColorInputProps, 'disabled' | 'gro
     },
   ({disabled}) => ({
     backgroundColor: disabled ? colors.soap200 : '',
+  }),
+  ({theme}) => ({
+    paddingLeft:
+      theme.direction === ContentDirection.LTR ? '46px' : 'calc(100% - 86px) /* @noflip */',
+    // We're using @noflip because ColorInput should stay LTR, therefore, we need to adjust the padding using ContentDirection, not using rtl-css-js.
   })
 );
 
-const ColorInputContainer = styled('div')({
-  position: 'relative',
-});
+const ColorInputContainer = styled('div')<Pick<ColorInputProps, 'grow'>>(
+  {
+    position: 'relative',
+    width: colorInputWidth,
+  },
+  ({grow}) =>
+    grow && {
+      minWidth: '100%',
+      width: '100%',
+    }
+);
 
 const PoundSignPrefix = styled('span')<Pick<ColorInputProps, 'disabled'>>(
   {
     position: 'absolute',
-    left: 36,
     top: 10,
     ...type.body,
-    ...type.variant.hint,
+    ...type.variant.mono,
+    color: colors.blackPepper300,
   },
   ({disabled}) => ({
     color: disabled ? inputColors.disabled.text : undefined,
+  }),
+  ({theme}) => ({
+    left: theme.direction === ContentDirection.LTR ? '36px' : '88px',
```

Comment:
Re. https://github.com/Workday/canvas-kit/pull/337#pullrequestreview-333591564, any time we do a special check like this we're going to have issues if the theme is undefined. For example, if I go into the storybook knob and set `direction` to an empty string (easy to do without typescript), this will turn into `left: 88px`.

Not sure of the best way to fix this. Maybe we explicitly set theme.direction to `ContentDirection.LTR` in [createCanvasTheme](https://github.com/Workday/canvas-kit/blob/master/modules/_labs/core/react/lib/theming/createCanvasTheme.ts#L80) if `partialTheme.direction` is not `ContentDirection.LTR` or `ContentDirection.RTL`

Reply:
Ok, should we do that in another branch?

Reply:
This is the only code affected by the RTL stuff so it's probably fine to do here